appinventor/appengine/src/com/google/appinventor/client/explorer/folder/ProjectFolder.java

```diff
@@ -264,6 +264,19 @@ public List<Project> getVisibleProjects(boolean onlySelected) {
     return list;
   }
 
+  public boolean containsAnyProjects() {
+    if (projectListItems.size() > 0) {
+      return true;
+    } else if (hasChildFolders()) {
+      for (ProjectFolder f : folders.values()) {
+        if (f.containsAnyProjects()) {
+          return true;
+        }
+      }
+    }
```

Comment:
Note that since `folders` should never be null, you could just make this an `else` and iterate over the (potentially empty) `folders`. Doing so doesn't require introducing another method in `hasChildFolders`